python/lsst/obs/base/gen2to3/convertRepo.py

```diff
@@ -459,7 +464,12 @@ def run(self, root: str, collections: List[str], *,
         # DM-21246 should let us fix this, assuming we actually want to keep
         # the transaction open that long.
         if self.config.doRegisterInstrument:
-            self.instrument.register(self.registry)
+            # Allow registration to fail on the assumption that this means
```

Comment:
I think I should be a bit cleverer here and run queryDimension and see if the instrument has been registered already.

Reply:
I think I'd vote to defer this until we can use `Database.sync` to do it via a new `Registry` method (see other comment on this topic); just doing `queryDimensions` yourself would be subject to race conditions, and while that probably isn't a big deal now, it means it definitely wouldn't be future-proof.

Reply:
I had implemented the queryDimension fix already. I think I'll leave it in for now since for gen2to3 the primary use case is a one to one conversion. Anyone converting multiple gen2 to one gen3 will likely be rare, especially doing it all in bulk simultaneously.